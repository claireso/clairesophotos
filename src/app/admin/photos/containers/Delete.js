import { connect } from 'react-redux'
import { navigate } from '@reach/router'

import Delete from '../views/Delete'

import {
  deletePhoto,
  DELETE_PHOTO_SUCCESS
} from '@common/actions/photos'

const mapDispatchToProps = dispatch => ({
  deletePhoto(id) {
    dispatch(deletePhoto(Number(id))).then(action => {
      if (action.type === DELETE_PHOTO_SUCCESS) {
        navigate('/admin/photos')
        return
      }

      document.querySelector('#modal').scrollTo(0, 0)
    })
  }
})

export default connect(
  null,
  mapDispatchToProps
)(Delete)
