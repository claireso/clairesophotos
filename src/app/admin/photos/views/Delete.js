import React from 'react'
import PropTypes from 'prop-types'

import Modal from '../../components/Modal'
import { PrimaryButton, SecondaryButton } from '../../components/Buttons'
import { Heading1 } from '../../components/Headings'
import Text from '../../components/Text'

const Delete = props => {
  return (
    <Modal onClose={() => props.navigate('/admin/photos')}>
      <Heading1>Are you sure?</Heading1>
      <p>This action is irreversible</p>
      <Text align="right">
        <SecondaryButton
          onClick={event => {
            event.preventDefault()
            props.navigate('/admin/photos')
          }}
        >
          {' '}
          Cancel{' '}
        </SecondaryButton>
        <PrimaryButton
          onClick={event => {
            event.preventDefault()
            props.deletePhoto(props.id)
          }}
        >
          {' '}
          Yes{' '}
        </PrimaryButton>
      </Text>
    </Modal>
  )
}

Delete.propTypes = {
  navigate: PropTypes.func.isRequired,
  deletePhoto: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired
}

export default Delete
