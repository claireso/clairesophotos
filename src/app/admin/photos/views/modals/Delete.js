import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

import { PrimaryButton, SecondaryButton } from '@admin/components/Buttons'
import { Heading1 } from '@admin/components/Headings'
import Text from '@admin/components/Text'

const Delete = props => {
  const { isProcessing } = props
  return (
    <Fragment>
      <Heading1>Are you sure?</Heading1>
      <p>This action is irreversible</p>
      <Text align="right">
        <SecondaryButton
          onClick={event => {
            event.preventDefault()
            props.onClose()
          }}
        >
          {' '}
          Cancel{' '}
        </SecondaryButton>
        <PrimaryButton
          onClick={event => {
            event.preventDefault()
            if (isProcessing) return
            props.deletePhoto(props.id)
          }}
          isLoading={isProcessing}
        >
          {' '}
          Yes{' '}
        </PrimaryButton>
      </Text>
    </Fragment>
  )
}

Delete.propTypes = {
  deletePhoto: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
  isProcessing: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired
}

export default Delete
