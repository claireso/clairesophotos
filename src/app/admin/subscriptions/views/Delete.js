import React from 'react'
import PropTypes from 'prop-types'

import Modal from '../../components/Modal'
import { ButtonLink } from '../../components/Links'

const Delete = props => {
  return (
    <Modal>
      <h1>Are you sure?</h1>
      <p>This action is irreversible</p>
      <p className="align-right">
        <ButtonLink
          label="Cancel"
          className="btn--gray"
          onClick={event => {
            event.preventDefault()
            props.navigate('/admin/subscriptions')
          }}
        />
        <ButtonLink
          label="Yes"
          onClick={event => {
            event.preventDefault()
            props.deleteSubscription(props.id)
          }}
        />
      </p>
    </Modal>
  )
}

Delete.propTypes = {
  id: PropTypes.string.isRequired,
  navigate: PropTypes.func.isRequired,
  deleteSubscription: PropTypes.func.isRequired
}

export default Delete
