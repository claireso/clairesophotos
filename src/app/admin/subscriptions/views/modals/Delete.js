import React from 'react'
import PropTypes from 'prop-types'

import Modal from '@admin/components/Modal'
import { PrimaryButton, SecondaryButton } from '@admin/components/Buttons'
import { Heading1 } from '@admin/components/Headings'
import Text from '@admin/components/Text'

const Delete = ({ isProcessing, ...props }) => {
  return (
    <Modal onClose={() => props.onClose()}>
      <Heading1>Are you sure?</Heading1>
      <p>This action is irreversible</p>
      <Text align="right">
        <SecondaryButton
          onClick={event => {
            event.preventDefault()
            props.onClose()
          }}
        >
          Cancel
        </SecondaryButton>
        <PrimaryButton
          onClick={event => {
            event.preventDefault()
            if (isProcessing) return
            props.deleteSubscription(props.id)
          }}
          isLoading={isProcessing}
        >
          Yes
        </PrimaryButton>
      </Text>
    </Modal>
  )
}

Delete.propTypes = {
  id: PropTypes.number.isRequired,
  deleteSubscription: PropTypes.func.isRequired,
  isProcessing: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired
}

export default Delete
