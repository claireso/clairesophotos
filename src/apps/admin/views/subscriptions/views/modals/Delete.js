import React, { Fragment, useCallback } from 'react'
import PropTypes from 'prop-types'

import { PrimaryButton, SecondaryButton } from '@admin/components/Buttons'
import { Heading1 } from '@admin/components/Headings'
import Text from '@admin/components/Text'

const Delete = ({ isProcessing, id, deleteSubscription, onClose }) => {
  const onCancel = useCallback(
    event => {
      event.preventDefault()
      onClose()
    },
    [onClose]
  )

  const onConfirm = useCallback(
    event => {
      event.preventDefault()
      if (isProcessing) return
      deleteSubscription(id)
    },
    [isProcessing, id, deleteSubscription]
  )

  return (
    <Fragment>
      <Heading1>Are you sure?</Heading1>
      <p>This action is irreversible</p>
      <Text align="right">
        <SecondaryButton onClick={onCancel}>Cancel</SecondaryButton>
        <PrimaryButton onClick={onConfirm} isLoading={isProcessing}>
          Yes
        </PrimaryButton>
      </Text>
    </Fragment>
  )
}

Delete.propTypes = {
  id: PropTypes.number.isRequired,
  deleteSubscription: PropTypes.func.isRequired,
  isProcessing: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired
}

export default Delete
