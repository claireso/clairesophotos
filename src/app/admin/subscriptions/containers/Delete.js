import { connect } from 'react-redux'
import { navigate } from '@reach/router'

import Delete from '../views/Delete'

import {
  deleteSubscription,
  DELETE_SUBSCRIPTION_SUCCESS
} from '../../../../common/actions/subscriptions'

const mapDispatchToProps = dispatch => ({
  deleteSubscription(id) {
    dispatch(deleteSubscription(Number(id))).then(action => {
      if (action.type === DELETE_SUBSCRIPTION_SUCCESS) {
        navigate('/admin/subscriptions')
        return
      }

      document.querySelector('.modal').scrollTo(0, 0)
    })
  }
})

export default connect(
  null,
  mapDispatchToProps
)(Delete)
