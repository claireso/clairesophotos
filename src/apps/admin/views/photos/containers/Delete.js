import { connect } from 'react-redux'
import { navigate } from '@reach/router'
import qs from 'qs'

import Delete from '../views/modals/Delete'

import { deletePhoto, DELETE_PHOTO_SUCCESS } from '@admin/actions/photos'
import { displaySuccessMessage } from '@admin/actions/messages'

const mapStateToProps = state => ({
  isProcessing: state.photos.isProcessing
})

const mapDispatchToProps = dispatch => ({
  deletePhoto(id) {
    dispatch(deletePhoto(Number(id))).then(action => {
      if (action.type === DELETE_PHOTO_SUCCESS) {
        const query = qs.parse(window.location.search.substring(1))
        const search = qs.stringify({
          ...query,
          action: undefined,
          id: undefined
        })
        navigate(`?${search}`)
        dispatch(
          displaySuccessMessage({
            message: 'Your photo has been deleted successfully',
            key: 'CRUD_PHOTO'
          })
        )
        window.scrollTo(0, 0)
        return
      }

      document.querySelector('#modal').scrollTo(0, 0)
    })
  }
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Delete)
