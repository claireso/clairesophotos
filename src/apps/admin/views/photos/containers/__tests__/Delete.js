import React from 'react'
import { Provider } from 'react-redux'
import { render, fireEvent, wait } from '@testing-library/react'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import Delete from '../Delete'
import callApiMiddleware from '@admin/middleware/callApi'

jest.mock('axios', () => {
  return jest.fn().mockImplementation(({ url }) => {
    switch (url) {
      case '/photos/1':
        return Promise.resolve({ data: {} })

      default:
        return Promise.resolve({ data: {} })
    }
  })
})

const middlewares = [thunk, callApiMiddleware]
const mockStore = configureMockStore(middlewares)

describe('<Delete />', () => {
  const renderComponent = () => {
    const store = mockStore({ user: {}, photos: { isProcessing: false } })

    const rendered = render(
      <Provider store={store}>
        <Delete onClose={() => {}} id={1} />
      </Provider>
    )

    return {
      ...rendered,
      store
    }
  }

  test('should render component', () => {
    const { container } = renderComponent()

    expect(container).toMatchSnapshot()
  })

  test('should delete photo', async () => {
    const { getByText, store } = renderComponent()

    fireEvent.click(getByText('Yes'))

    const expectedActions = [
      { type: 'DELETE_PHOTO_REQUEST', id: 1 },
      { type: 'DELETE_PHOTO_SUCCESS', response: {}, id: 1 },
      {
        type: 'ADD_MESSAGE',
        status: 'success',
        key: 'CRUD_PHOTO',
        message: 'Your photo has been deleted successfully'
      }
    ]

    await wait(() => {
      expect(store.getActions()).toEqual(expectedActions)
    })
  })
})
