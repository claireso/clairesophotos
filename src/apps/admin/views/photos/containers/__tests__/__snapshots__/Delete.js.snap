// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Delete /> should render component 1`] = `
.c5 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  border-radius: 0.2rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Roboto',Arial,sans-serif;
  font-size: 1.2rem;
  line-height: 1.3333333;
  outline: none;
  padding: 1.6rem;
  -webkit-transition: background 150ms ease-out;
  transition: background 150ms ease-out;
  background: var(--primary);
  color: white;
}

.c5 > svg {
  margin: 0 -0.4rem 0 0.4rem;
}

.c5 + .c4 {
  margin: 0 0 0 0.8rem;
}

.c5:hover {
  background: var(--primary-lighter);
}

.c3 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  border-radius: 0.2rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Roboto',Arial,sans-serif;
  font-size: 1.2rem;
  line-height: 1.3333333;
  outline: none;
  padding: 1.6rem;
  -webkit-transition: background 150ms ease-out;
  transition: background 150ms ease-out;
  background: var(--secondary);
  color: var(--text);
}

.c3 > svg {
  margin: 0 -0.4rem 0 0.4rem;
}

.c3 + .c2 {
  margin: 0 0 0 0.8rem;
}

.c3:hover {
  background: var(--secondary-darker);
}

.c3 + .c4 {
  margin-left: 0.8rem;
}

.c0 {
  font-size: 2.4rem;
  margin: 0 0 2.4rem;
}

.c1 {
  text-align: right;
  font-size: 1.2rem;
}

<div>
  <h1
    class="c0"
  >
    Are you sure?
  </h1>
  <p>
    This action is irreversible
  </p>
  <p
    class="c1"
  >
    <button
      class="c2 c3"
    >
       Cancel 
    </button>
    <button
      class="c4 c5"
    >
       
      Yes
       
    </button>
  </p>
</div>
`;
